src/main/java/com/aws/greengrass/clientdevices/auth/connectivity/CISShadowMonitor.java

```diff
@@ -33,45 +33,88 @@
 import java.util.concurrent.Future;
 import java.util.concurrent.TimeUnit;
-import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
 import java.util.function.Consumer;
+import java.util.function.Supplier;
 import javax.inject.Inject;
 
 @SuppressWarnings("PMD.ImmutableField")
 public class CISShadowMonitor implements Consumer<NetworkStateProvider.ConnectionState> {
     private static final Logger LOGGER = LogManager.getLogger(CISShadowMonitor.class);
     private static final String CIS_SHADOW_SUFFIX = "-gci";
     private static final String VERSION = "version";
-    private static final long TIMEOUT_FOR_SUBSCRIBING_TO_TOPICS_SECONDS = Duration.ofMinutes(1).getSeconds();
-    private static final long WAIT_TIME_TO_SUBSCRIBE_AGAIN_IN_MS = Duration.ofMinutes(2).toMillis();
-    private static final Random JITTER = new Random();
-    static final String SHADOW_UPDATE_DELTA_TOPIC = "$aws/things/%s/shadow/update/delta";
-    static final String SHADOW_GET_ACCEPTED_TOPIC = "$aws/things/%s/shadow/get/accepted";
 
+    private static final RetryUtils.RetryConfig SUBSCRIBE_TO_TOPICS_RETRY_CONFIG =
+            RetryUtils.RetryConfig.builder()
+                    .initialRetryInterval(Duration.ofSeconds(1L))
+                    .maxRetryInterval(Duration.ofSeconds(30L))
+                    .maxAttempt(Integer.MAX_VALUE)
+                    .retryableExceptions(Collections.singletonList(Exception.class))
+                    .build();
     private static final RetryUtils.RetryConfig GET_CONNECTIVITY_RETRY_CONFIG =
             RetryUtils.RetryConfig.builder().initialRetryInterval(Duration.ofMinutes(1L))
                     .maxRetryInterval(Duration.ofMinutes(30L)).maxAttempt(Integer.MAX_VALUE)
                     .retryableExceptions(Arrays.asList(ThrottlingException.class, InternalServerException.class))
                     .build();
+    private static final RetryUtils.RetryConfig GET_CIS_SHADOW_RETRY_CONFIG =
+            RetryUtils.RetryConfig.builder()
+                    .initialRetryInterval(Duration.ofSeconds(1L))
+                    .maxRetryInterval(Duration.ofSeconds(30L))
+                    .maxAttempt(Integer.MAX_VALUE)
+                    .retryableExceptions(Collections.singletonList(Exception.class))
+                    .build();
+
+    // we only need to subscribe to shadow topics once across the lifetime of this monitor
+    private final AtomicBoolean subscribed = new AtomicBoolean();
+
+    // we don't need to unsubscribe, because mqtt client reconnects with clean session on startup.
+    // instead, we can cancel current operations and block new ones on shutdown.
+    private final AtomicBoolean stopped = new AtomicBoolean();
+    private final Consumer<ShadowDeltaUpdatedEvent> onShadowDeltaUpdated = event -> {
+        if (stopped.get()) {
+            return;
+        }
+        processCISShadow(event);
```

Comment:
if we got an event, why would we ignore it?

Reply:
this was to handle shutdown w/o unsubscribe

Reply:
will add unsubscribe back in, looks to be less confusing---was originally thinking on the terms of not having extra mqtt calls if after shutdown we'd lose the subscription anyway (since nucleus first-time connect is w/ clean session)

Reply:
If we're shutting down, why does it matter then if we process the event or not?

Reply:
Was to avoid extra processing if we receive a shadow update after the task was already cancelled. 

I guess it's not a big deal though because nucleus will shutdown the executors eventually